aws/sdk/examples/ec2/src/bin/list-all-instance-events.rs

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use aws_types::region::ProvideRegion;
+use ec2::{Client, Config, Error, Region, PKG_VERSION};
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+struct Opt {
+    /// The AWS Region in which the client is created.
+    #[structopt(short, long)]
+    region: Option<String>,
+
+    /// Whether to display additional runtime information.
+    #[structopt(short, long)]
+    verbose: bool,
+}
+
+/// Shows the scheduled events for the instances in the Region.
+async fn show_events(reg: String) {
```

Comment:
To avoid unnecessary memory copies, you'll want to make this take a `str` reference:
```rust
async fn show_events(reg: &str) {
    // Note: the clone isn't necessary now since Region::new 
    // takes `impl Into<Cow<'static, str>>` which converts it into the
    // type it needs automatically
    let region = Region::new(reg);
```

Then when you call `show_events`, you'll need to convert it to a `str` reference:
```rust
show_events(region.region_name.as_ref().unwrap()).await;
```

Reply:
I'm getting the following error from the `let region = Region::new(reg)` line:
```
`reg` has an anonymous lifetime `'_` but it needs to satisfy a `'static` lifetime requirement

...is captured here...
```